docs/source/Compiler.md

```diff
@@ -127,6 +127,7 @@ Additional options:
 
 -   `--reflect-types` : Add minimal type reflection to code generation.
 -   `--reflect-names` : Add minimal type/name reflection.
+-   `--reflect-attrs` : Add reflection for user-defined attributes.
```

Comment:
Maybe `Add minimal type/name/attribute reflection` to make it obvious this is incremental.

Reply:
I am not sure how I feel about this because it seems the decision for these to be incremental is arbitrary and not out of necessity. Every possible use case people will have for reflection is hard to define and thus you can't really deduce how much type information they are going to need. In my particular use case, it just happens that I do in fact need all 3, but I believe there are potentially many use cases where you may only need the attributes and nothing else.

I truly believe that the most used option is going to be `--reflect-attrs` because people want to use them and there's only two other ways (full reflection or linking the parser), both of which are cumbersome. This is why I do not really want to see it conflated with the other command line options.

I see two potential solutions here that I think you should consider:
* Add more short options to make it easier to combine them.
* Add an extra option that does combine all three i.e. `--reflect-full`.